raft/server.cc

```diff
@@ -1678,12 +1690,13 @@ future<> server_impl::set_configuration(config_member_set c_new, seastar::abort_
 
     auto f = _non_joint_conf_commit_promise.emplace().promise.get_future();
     if (as) {
-        _non_joint_conf_commit_promise->abort = as->subscribe([this] () noexcept {
+        _non_joint_conf_commit_promise->abort = as->subscribe([this, e] () noexcept {
             // If we're inside this callback, the subscription wasn't destroyed yet.
             // The subscription is destroyed when the field is reset, so if we're here, the field must be engaged.
             SCYLLA_ASSERT(_non_joint_conf_commit_promise);
             // Whoever resolves the promise must reset the field. Thus, if we're here, the promise is not resolved.
-            std::exchange(_non_joint_conf_commit_promise, std::nullopt)->promise.set_exception(request_aborted{});
+            std::exchange(_non_joint_conf_commit_promise, std::nullopt)->promise.set_exception(
+                request_aborted(format("Aborted while setting configuration (at index: {}, term: {}, config: {})", e.idx, e.term, _fsm->get_configuration())));
```

Comment:
`config` -> `current config` to make it more precise (otherwise, the reader might get confused and misunderstand that the config refers to the one at `e.idx` -- which doesn't need to be true)